### PR TITLE
Support zone redundancy on ASPs

### PR DESCRIPTION
**Pull request by nargiz**
This PR closes #33698

The changes in this PR are as follows:

* Introduced `enable_zone_redundant` for WebApp/ServicePlan

I have read the [contributing guidelines](CONTRIBUTING.md) and have completed the following:

* [x] **Tested my code** end-to-end against a live Azure subscription.
* [x] **Updated the documentation** in the docs folder for the affected changes.
* [x] **Written unit tests** against the modified code that I have made.
* [x] **Updated the [release notes](RELEASE_NOTES.md)** with a new entry for this PR.
* [x] **Checked the coding standards** outlined in the [contributions guide](CONTRIBUTING.md) and ensured my code adheres to them.

If I haven't completed any of the tasks above, I include the reasons why here:

Below is a minimal example configuration that includes the new features, which can be used to deploy to Azure:

```fsharp

let myasp = servicePlan {
    name "zone-redundant-sp"
    sku WebApp.Sku.P1V2
    enable_zone_redundant
}
```

OR 

```fsharp

let myapp = webApp{
    name "zone-redundant-app"
    sku WebApp.Sku.P1V2
    enable_zone_redundant
}
```

### CODE REVIEW

File: src/Farmer/Builders/Builders.WebApp.fs [hunk 1]
```diff
@@ -603,7 +605,8 @@ type WebAppBuilder() =
           AutomaticLoggingExtension = true
           SiteExtensions = Set.empty
           PrivateEndpoints = Set.empty
-          CustomDomains = Map.empty }
+          CustomDomains = Map.empty
+          ZoneRedundant = None }
```
Comment (BlakeWills) [inline src/Farmer/Builders/Builders.WebApp.fs:609]: Should `None` be the default or `false`? What happens if you build a server farm without setting `enable_zone_redundant`?

File: src/Tests/ServicePlan.fs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+open Farmer
+open Farmer.Builders
+open Farmer.Arm
+open System
+
+let getResource<'T when 'T :> IArmResource> (data:IArmResource list) = data |> List.choose(function :? 'T as x -> Some x | _ -> None)
+let getResources (v:IBuilder) = v.BuildResources Location.WestEurope
+
+let tests = testList "Service Plan Tests"[
+    test "Enable zoneRedundant in service plan" {
```
Comment (BlakeWills) [inline src/Tests/ServicePlan.fs:13]: Do we also want to add a test to cover the default? (When you don't enable it the value is x?)

File: src/Farmer/Builders/Builders.WebApp.fs (hunk 1 above)
Comment (nargiz) [reply src/Farmer/Builders/Builders.WebApp.fs:609]: Farmer already had zone redundancy for EventHub, there it was kinda nullable. Tried to keep it consistent. 
When we don't specify it, farmer doesn't include ZoneRedundant property in ARM template and by default Azure doesn't make it ZoneRedundant. I will give a test with ZoneRedundant false in ARM template and let you know

Comment (BlakeWills) [reply src/Farmer/Builders/Builders.WebApp.fs:609]: Happy to keep it consistent with EventHub, I just wanted to make sure we haven't broken anything that doesn't specify zone redundancy :)

**Comment by nargiz** [comment]
This feature is merged to CompositionalIT repo

**Pull request closed without merging**